open_city_profile/graphene.py

```diff
@@ -70,3 +77,57 @@ def resolve(self, next, root, info, **kwargs):
             self.cached_loaders = True
 
         return next(root, info, **kwargs)
+
+
+TranslationLanguageType = graphql.GraphQLEnumType(
+    name="TranslationLanguage",
+    values=OrderedDict(
+        {
+            lang[0].upper(): graphql.GraphQLEnumValue(lang[0], description=lang[1])
+            for lang in settings.LANGUAGES
+        }
+    ),
+)
+
+
+TranslationDirective = graphql.GraphQLDirective(
+    name="translation",
+    locations=[graphql.DirectiveLocation.FIELD],
+    args={
+        "in": graphql.GraphQLArgument(
+            graphql.GraphQLNonNull(TranslationLanguageType), description="The language"
+        ),
+    },
+    description="Get value translated in language",
+)
+
+
+class TranslationMiddleware:
+    def resolve(self, next, root, info, **kwargs):
+        directives = info.field_asts[0].directives
+
+        if not directives:
+            return next(root, info, **kwargs)
+
+        translation_directives = [
+            directive
+            for directive in directives
+            if directive.name == Name("translation")
+        ]
+        if not translation_directives:
+            return next(root, info, **kwargs)
+
+        directive = translation_directives[0]
+        language = graphql.value_from_ast(
+            directive.arguments[0].value, TranslationLanguageType, info.variable_values
```

Comment:
The more correct, albeit longer, way to get the `in` argument could be for example `next(arg for arg in directive.arguments if arg.name.value == "in")`.

Reply:
The arguments are validated so I don't think either way is more correct?

Reply:
I mean "the correct logic". The correct logic is something like "the argument called `in` specifies the language". Not "the first argument specifies the language". GraphQL has only named arguments, not positional. Here the difference is of course small, because there is currently only one argument and it's mandatory. But the current implementation might fail if ever there will be another argument for the directive.

Reply:
Yeah, I guess that's true. I'll change it.

Just noticed that Graphene uses built-in shadowing `next` name as a parameter in their middlewares. I think I'll change that to something else.